src/models/participant/facade.js

```diff
@@ -47,6 +51,130 @@ const getByNameAndCurrency = async (name, currencyId) => {
   }
 }
 
+/**
+ * @function GetEndpoint
+ *
+ * @async
+ * @description This retuns the active endpoint value for a give participantId and type of endpoint
+ *
+ *
+ * @param {integer} participantId - the id of the participant in the database. Example 1
+ * @param {string} type - the type of the endpoint. Example 'FSIOP_CALLBACK_URL'
+ *
+ * @returns {array} - Returns participantEndpoint array containing the details of active endpoint for the participant if successful, or throws an error if failed
+ */
+
+const getEndpoint = async (participantId, endpointType) => {
+  try {
+    return Db.participantEndpoint.query(builder => {
+      return builder.innerJoin('endpointType AS et', 'participantEndpoint.endpointTypeId', 'et.endpointTypeId')
+        .where({
+          'participantEndpoint.participantId': participantId,
+          'participantEndpoint.isActive': 1,
+          'et.name': endpointType
+        }).select('participantEndpoint.participantEndpointId',
+          'participantEndpoint.participantId',
+          'participantEndpoint.endpointTypeId',
+          'participantEndpoint.value',
+          'participantEndpoint.isActive',
+          'participantEndpoint.createdDate',
+          'participantEndpoint.createdBy',
+          'et.name')
+    })
+  } catch (err) {
+    throw new Error(err.message)
+  }
+}
+
+/**
+ * @function GetAllEndpoints
+ *
+ * @async
+ * @description This retuns all the active endpoints for a give participantId
+ *
+ *
+ * @param {integer} participantId - the id of the participant in the database. Example 1
+ *
+ * @returns {array} - Returns an array containing the list of all active endpoints for the participant if successful, or throws an error if failed
+ */
+
+const getAllEndpoints = async (participantId) => {
+  try {
+    return Db.participantEndpoint.query(builder => {
+      return builder.innerJoin('endpointType AS et', 'participantEndpoint.endpointTypeId', 'et.endpointTypeId')
+        .where({
+          'participantEndpoint.participantId': participantId,
+          'participantEndpoint.isActive': 1
+        }).select('participantEndpoint.participantEndpointId',
```

Comment:
these are all the columns could you not just use .select('participantEndpointId.*', 'et.name')